project/default.nix

```diff
@@ -1,221 +1,276 @@
-  # the nix-shell.
+          Some tools, like `ghc-mod`, have to be built with the same GHC as
+          your project. The argument to the `tools` function is the haskell
+          package set of the platform we are developing for, allowing you to
+          build tools with the correct Haskell package set.
+  
+          The second argument, `super`, is the record of tools provided by
+          default. You can override these defaults by returning values with
+          the same name in your record. They can be disabled by setting them
+          to null.
+        '';
+        type = unspecified;
+        example = literalExample ''
+          ghc: super: {
+            inherit (ghc) hpack;
+            inherit (pkgs) chromium;
+            ghc-mod = null;
+            cabal-install = ghc.callHackage "cabal-install" "2.0.0.1" {};
+            ghcid = pkgs.haskell.lib.justStaticExecutables super.ghcid;
+          };
+        '';
+        default = _: _: {};
+      };
 
-, useWarp ? false
-  # Configure `reflex-dom` to use `jsaddle-warp`.
+      tools = mkOption {
+        description = ''
+          A function returning the list of tools to provide in the
+          nix-shells.  Some tools, like `ghc-mod`, have to be built
+          with the same GHC as your project. The argument to the
+          `tools` function is the haskell package set of the platform
+          we are developing for, allowing you to build tools with the
+          correct Haskell package set.
+        '';
+        type = unspecified;
+        example = literalExample ''
+          ghc: with ghc; [
+            hpack
+            pkgs.chromium
+          ]
+        '';
+        default = _: [];
+      };
 
-, android ? {}
-  # ::
-  # { <app name> ::
-  #   { executableName :: String
-  #   , applicationId :: String
-  #   , displayName :: String
-  #   , package :: PackageSet -> Derivation
-  #     ^ Optional
-  #   }
-  # }
-  #
-  # Use this argument to configure android apps. The returned
-  # derivations will be in `android.<app name>`. The `package`
-  # argument can be set to use a different Haskell package than the
-  # one named <app name>.
+      withHoogle = mkOption {
+        description = ''
+          Set to false to disable building the hoogle database when
+          entering the nix-shell.
+        '';
+        type = bool;
+        default = true;
+      };
 
-, ios ? {}
-  # ::
-  # { <app name> ::
-  #   { executableName :: String
-  #   , bundleIdentifier :: String
-  #   , bundleName :: String
-  #   , package :: PackageSet -> Derivation
-  #     ^ Optional
-  #   }
-  # }
-  #
-  # Use this argument to configure iOS apps. The returned derivations
-  # will be in `ios.<app name>`. The `package` argument can be set to
-  # use a different Haskell package than the one named <app name>.
+      useWarp = mkOption {
+        description = ''
+          Configure `reflex-dom` to use `jsaddle-warp`.
```

Comment:
@ElvishJerricco even I am not sure what exactly this option is doing, or how to use this.

Reply:
If you set it, an override is included which configures `reflex-dom` with `use-warp`.